Extension/src/LanguageServer/client.ts

```diff
@@ -521,22 +577,54 @@ class DefaultClient implements Client {
                 let settings: CppSettings = new CppSettings(this.RootUri);
                 this.languageClient.sendNotification(UpdateIntelliSenseCachePathNotification, util.resolveVariables(settings.intelliSenseCachePath, this.AdditionalEnvironment));
             }
+            if (changedSettings["textMateColorization"]) {
+                this.updateGrammarsInPackageJson();
+            }
             this.configuration.onDidChangeSettings();
             telemetry.logLanguageServerEvent("CppSettingsChange", changedSettings, null);
         }
 
+        let settings: CppSettings = new CppSettings(this.RootUri);
+        if (settings.enhancedColorization) {
+            this.colorizationSettings.reload();
+        }
+
+        // Update colorization
+        let processedUris: vscode.Uri[] = [];
+        for (let e of vscode.window.visibleTextEditors) {
+            let uri: vscode.Uri = e.document.uri;
+            // Make sure we don't process the same file multiple times
+            if (!processedUris.find(e => e === uri)) {
```

Comment:
Does this work when a file is split in two windows?

Reply:
Yes.  (?).  This is calling onSettingsChanged for each file visible in an editor, but it should not do so multiple times for the same file.  onSettingsChanged ensures all visible text editors for that file get refreshed, after it's created the (same) set of decorators for that file.  (The set of decorators are shared by all visible instances of the file).